src/execution_plan/ops/op_cond_var_len_traverse.c

```diff
@@ -89,10 +85,18 @@ OpBase *NewCondVarLenTraverseOp(const ExecutionPlan *plan, Graph *g, AlgebraicEx
 	// Populate edge value in record only if it is referenced.
 	AST *ast = QueryCtx_GetAST();
 	QGEdge *e = QueryGraph_GetEdgeByAlias(op->qg, AlgebraicExpression_Edge(op->ae));
-	if(AST_AliasIsReferenced(ast, e->alias))
-		op->edgesIdx = OpBase_Modifies((OpBase *)op, e->alias);
+	if(AST_AliasIsReferenced(ast, e->alias)) op->edgesIdx = OpBase_Modifies((OpBase *)op, e->alias);
 	else op->edgesIdx = -1;
 
+	if(e->bidirectional) {
+		op->traverseDir = GRAPH_EDGE_DIR_BOTH;
+	} else if(AlgebraicExpression_Transposed(ae)) {
+		// If the sole operand in the AlgebraicExpression is transposed, we are traversing right-to-left.
+		op->traverseDir = GRAPH_EDGE_DIR_INCOMING;
+	} else {
+		op->traverseDir = GRAPH_EDGE_DIR_OUTGOING;
+	}
```

Comment:
see if this fits as a one liner
op->traverseDir = AlgebraicExpression_Transposed(ae) ? GRAPH_EDGE_DIR_INCOMING : GRAPH_EDGE_DIR_OUTGOING;

Reply:
It does not, sorry!